packages/pyolite-kernel/src/worker.ts

```diff
@@ -412,6 +419,10 @@ self.onmessage = async (event: MessageEvent): Promise<void> => {
       results = commClose(messageContent);
       break;
 
+    case 'set-interrupt-buffer':
```

Comment:
hm... i don't know if the intent of this API was to closely map to the kernel messaging spec. Ideally, there would be some way to model this on top of jupyter primitives instead of adding a new message type.

Reply:
This is pretty low level, and i'm not sure it's well suited for the comms API.  Especially since I'm not sure what would happen if you passed a SharedArrayBuffer to python and back. I can certainly test that though, if you think we should.

Reply:
Right, there are some limits to what will work (and across what boundaries, see #463) and not everything is going to map perfectly, but as the first "custom" message on this pipe, it's worth taking a look. If we _could_ use these buffers directly, that might be _great_ for e.g. #195.

Perhaps we need to further differentiate the types of messages... it would be lovely for us to be able to: first distinguish between:
- "well-known" kernel message types from JupyterLab
  - ... and be able to use more typescript types from upstream, and maybe (ab)use totality checks in the `switch`
- "custom" jupyterlite ones
  - there might be a class of ones that _all_ lite kernels might need, e.g. _what is the root of the lite site_
  - per-kernel ones...

Reply:
I'm not sure where to go from here. Do you have any concrete suggestions on an approach?